src/Particle/Par_Aux_Check_Particle.cpp

```diff
@@ -72,293 +97,546 @@ void Par_Aux_Check_Particle( const char *comment )
+
+// tracer particles must be massless
+#  ifdef TRACER
+   if ( amr->Par->Type[ParID] == PTYPE_TRACER  &&  amr->Par->Mass[ParID] != (real_par)0.0 )
+      CheckTypePass = false;
+#  endif
+
+   if ( CheckTypePass )   return;
+
+   if ( PassAll )
+      Aux_Message( stderr, "\"%s\" : <%s> FAILED at Time = %13.7e, Step = %ld !!\n",
+                   comment, __FUNCTION__, Time[lv], Step );
+
+   if ( PassOne )
+      Aux_Message( stderr, "Check 11: %4s  %2s  %7s  %10s  %10s  %20s\n",
+                   "Rank", "Lv", "PID", "ParID", "Type", "Mass" );
+
+   Aux_Message( stderr, "Check 11: %4d  %2d  %7d  %10ld  %10d  %20.13e\n",
+                MPI_Rank, lv, PID, ParID, (int)amr->Par->Type[ParID], amr->Par->Mass[ParID] );
+
+   PassAll = false;
+   PassOne = false;
+
+} // FUNCTION : Check_ValidType
+
+
+
+//-------------------------------------------------------------------------------------------------------
+// Function    :  Check_ValidUID
+// Description :  Check if particle has valid UID
+//
+// Note        :  None
+//
+// Parameter   :  PassAll  : Pass all the checkes or not
+//                PassOne  : Pass this check or not
+//                comment  : You can put the location where this function is invoked in this string
+//-------------------------------------------------------------------------------------------------------
+void Check_ValidUID( int &PassAll, int &PassOne, const char *comment )
+{
+
+   for (long p=0; p<amr->Par->NPar_AcPlusInac; p++)
+   {
+      if ( amr->Par->PUid[p] > (long_par)0  &&  amr->Par->PUid[p] < amr->Par->NextUID )  continue;
+
+      if ( PassAll )
+         Aux_Message( stderr, "\"%s\" : <%s> FAILED at Time = %13.7e, Step = %ld !!\n",
+                      comment, __FUNCTION__, Time[0], Step );
+
+      if ( PassOne )
+         Aux_Message( stderr, "Check 12: %4s  %10s  %10s  %10s\n", "Rank", "ParID", "PUid", "NextUID" );
+
+      Aux_Message( stderr, "Check 12: %4d  %10ld  %10ld  %10ld\n", MPI_Rank, p, (long)amr->Par->PUid[p], (long)amr->Par->NextUID );
+
+      PassAll = false;
+      PassOne = false;
+   } // for (long p=0; p<amr->Par->NPar_AcPlusInac; p++)
+
+} // FUNCTION : Check_ValidUID
+
+
+
+//-------------------------------------------------------------------------------------------------------
+// Function    :  Check_UniqueUID
+// Description :  Check if particle has unique UID
+//
+// Note        :  None
+//
+// Parameter   :  PassAll : Pass all the checkes or not
+//                PassOne : Pass this check or not
+//                comment : You can put the location where this function is invoked in this string
+//                Par_UID : Array of number of particle UID
+//-------------------------------------------------------------------------------------------------------
+void Check_UniqueUID( int &PassAll, int &PassOne, const char *comment, int *Par_UID )
+{
+
+   for (long p=0; p<amr->Par->NPar_AcPlusInac; p++)
+   {
+      if ( amr->Par->Mass[p] < (long_par)0.0 )   continue;
```

Comment:
Is this to exclude the inactive particles? I thought inactive particles also have a UID. And why does it use `long_par` for `0.0`?